app/scripts/controllers/reviewRegistration.js

```diff
@@ -280,4 +329,117 @@ angular.module('confRegistrationWebApp')
+    function mergeWithSpouse () {
+      // Generate an array of new registrants that include all attributes
+      var newRegistrants = $scope.currentRegistration.registrants.map(function (registrant) {
+        var newRegistrantId = uuid();
+
+        // Make a copy of the answers, but overwrite the id and registrantId attributes
+        var answers = registrant.answers.map(function(answer) {
+          return _.assign({}, answer, {
+            id: uuid(),
+            registrantId: newRegistrantId
+          });
+        });
+
+        return {
+          id: newRegistrantId,
+          registrationId: $scope.spouseRegistration.id,
+          registrantTypeId: registrant.registrantTypeId,
+          answers: answers,
+          firstName: registrant.firstName,
+          lastName: registrant.lastName,
+          email: registrant.email
+        };
+      });
+
+      //Payload for new spouse registration
+      var newSpouseRegistration = {
+        id: $scope.spouseRegistration.id,
+        conferenceId: $scope.currentRegistration.conferenceId,
+        registrants: newRegistrants.map(function (registrant) {
+          // When creating a new registration, only a few registrant attributes are required, so only keep a few of the
+          // registrant attributes
+          var sparseRegistrant = _.pick(registrant, ['id', 'registrationId', 'registrantTypeId']);
+          sparseRegistrant.answers = [];
+          return sparseRegistrant;
+        })
+      };
+
+      //Add new registration
+      return createRegistration(newSpouseRegistration)
+        .then(function () {
+          //Add registrants to new registration
+          // Advance to the next step after all the registrations have been created
+          return $q.all(newRegistrants.map(createRegistrant));
+        }).then(function () {
+          //Delete existing registration
+          return deleteRegistration($scope.currentRegistration);
+        }).then(function () {
+          // Reload the merged spouse registration
+          return loadRegistration($scope.spouseRegistration.id);
+        }).then(function (mergedRegistration) {
+          // Update the UI to show the merged registration because it includes all of the registrants
+          $scope.currentRegistration = mergedRegistration;
+
+          // Hide certain elements and sections in the UI because the current user is not able make changes to their
+          // spouses registration, even though they are now a registrant on that registration
+          $scope.mergedRegistration = true;
+        }).catch(errorFromResponse('An error occurred while merging spouse registrations.'));
+    }
+
+    // Called when the user clicks the register together button
+    $scope.mergeAndConfirmRegistration = function () {
+      // Complete the current registration before merging it with the spouse
+      $scope.submittingRegistration = true;
+      confirmRegistration().then(function () {
```

Comment:
If one spouse has already registered both of them and the other spouse attempts to register, it will detect this condition and display a warning message. The user can still register anyway, although the UI makes it clear that this is not recommended.  `$scope.alreadyRegistered` will be set to `true` in this case, so you can look at the template and see what exactly will be displayed.